starknet-foundry/crates/cast/src/helpers/scarb_utils.rs

```diff
@@ -31,30 +31,34 @@ fn get_property(
             )),
         None => profiled
             .and_then(|t| t.get(property))
-            .and_then(serde_json::Value::as_str)
+            .and_then(Value::as_str)
             .map(String::from)
             .ok_or(anyhow!("Property not found in tool: {property}")),
     }
 }
 
 pub fn parse_scarb_config(
     profile: &Option<String>,
-    path: Option<Utf8PathBuf>,
+    path: &Option<Utf8PathBuf>,
 ) -> Result<CastConfig> {
-    let manifest_path = find_manifest_path(path.as_ref().map(camino::Utf8PathBuf::as_path))
-        .expect("Failed to obtain Scarb.toml file path");
-
-    if let Some(manifest_path) = path {
-        if !manifest_path.exists() {
-            bail! {"{manifest_path} file does not exist!"};
-        };
+    let found_path = find_manifest_path(path.as_ref().map(Utf8PathBuf::as_path));
```

Comment:
I don't get why we need `path.as_ref()` when `path` is already a ref.

Reply:
I'm not an expert, but here is an error
![image](https://github.com/foundry-rs/starknet-foundry/assets/61014013/fc0c4487-8263-48e6-86cd-40479facd5ee)

Reply:
It's because
`path` - `&Option<Utf8PathBuf>`
`path.as_ref()` - `Option<&Utf8PathBuf>`